src/layer/rail.js

```diff
@@ -0,0 +1,356 @@
+"use strict";
+
+import * as Color from "../constants/color.js";
+import * as Util from "../js/util.js";
+
+// Exponent base for inter-zoom interpolation
+let railExp = 1.2;
+
+// Helper function to create a "filter" block for a particular railway class.
+function filterRail(railClass, service, brunnel) {
+  return [
+    "all",
+    brunnel === "surface"
+      ? ["!in", "brunnel", "bridge", "tunnel"]
+      : ["==", "brunnel", brunnel],
+    ["==", "class", "rail"],
+    ["==", "subclass", railClass],
+    [service ? "in" : "!in", "service", "siding", "spur", "yard"],
+  ];
+}
+
+// Base definition that applies to all railways
+var defRail = {
+  type: "line",
+  source: "openmaptiles",
+  "source-layer": "transportation",
+};
+
+// Generate a unique layer ID
+function uniqueLayerID(railClass, service, part, brunnel, constraints) {
+  var layerID = [railClass, service ? "service" : "rail", part, brunnel].join(
+    "_"
+  );
+  if (constraints != null) {
+    layerID +=
+      "_" + constraints.join("_").replaceAll("=", "").replaceAll("-", "_");
+  }
+  return layerID;
+}
+
+function baseRailLayer(railClass, id, brunnel, minzoom, service, constraints) {
+  var layer = Util.layerClone(
+    defRail,
+    uniqueLayerID(railClass, service, id, brunnel, constraints)
+  );
+  layer.filter = filterRail(railClass, service, brunnel);
+  layer.minzoom = minzoom;
+  return layer;
+}
+
+// Base railway class
+
+class Railway {
+  constructor() {
+    this.hue = 35;
+  }
+  bridgeCasing = function () {
+    var layer = baseRailLayer(
+      this.railClass,
+      "bridgeCasing",
+      this.brunnel,
+      this.minZoomBridge,
+      this.service,
+      this.constraints
+    );
+    layer.layout = {
+      "line-cap": "butt",
+      "line-join": "bevel",
+      visibility: "visible",
+    };
+    layer.paint = {
+      "line-color": "hsl(0, 0%, 50%)",
```

Comment:
At a glance, this casing looked like an abutting street to me.

Consider a knockout effect instead, as we explored in #169: no outer line, just a background that matches the land background. The knockout should be thin to avoid visually intruding on waterways or greenspace. However, it should be more than a hairline: #203.

A more literal representation of a bridge might require work that’d be out of scope for this PR. It seems like you were going for guardrails, but the guardrails would need to flare out at the abutments to avoid looking like road casings. It might be tricky to get those flares to line up with the guardrails.

In other maps, instead of guardrails, there are schematic-like break symbols that don’t necessarily extend the full length of the bridge. For example, the first photo in https://github.com/ZeLonewolf/openstreetmap-americana/issues/169#issuecomment-1043648465 depicts a bridge that actually extends all the way past the river crossing out of view.

Reply:
I like the knockout casing better:

![Screenshot from 2022-07-17 11-57-10](https://user-images.githubusercontent.com/1732117/179407935-92328f5d-f1d2-4018-9876-76b6d2bca9f9.png)

Reply:
that knockout looks great!